linchpin/linchpin.constants

```diff
@@ -11,10 +11,12 @@ rundb_conn_type = file
 rundb_schema = {"action": "",
                 "inputs": [],
                 "outputs": [],
+                "topology_outputs": [],
```

Comment:
@herlo 
Though the outputs and topology outputs seems to be same , There is a subtle difference between both:
```
(Pdb) import pprint
(Pdb) pprint.pprint(targets[name]["topology_outputs"])
{u'dummy_res': [{u'changed': False,
                 u'dummy_file': u'/tmp/dummy.hosts',
                 u'failed': False,
                 u'hosts': [u'web-fde56f-0.example.net']},
                {u'changed': False,
                 u'dummy_file': u'/tmp/dummy.hosts',
                 u'failed': False,
                 u'hosts': [u'web2-fde56f-0.example.net',
                            u'web2-fde56f-1.example.net']}]}
(Pdb) pprint.pprint(targets[name]["outputs"])
{u'resources': [{u'changed': False,
                 u'dummy_file': u'/tmp/dummy.hosts',
                 u'failed': False,
                 u'hosts': [u'web-fde56f-0.example.net']},
                {u'changed': False,
                 u'dummy_file': u'/tmp/dummy.hosts',
                 u'failed': False,
                 u'hosts': [u'web2-fde56f-0.example.net',
                            u'web2-fde56f-1.example.net']}]}
```
As you can see , topology_outputs segregates the resources based on the type of the resource that is being generated Which made the existing inventory generation code reusable. So in the above case if there is aws_ec2 resource provisioned inside the topology  along with dummy topology their respective topology_outputs and outputs looks as follows:
```
(Pdb) import pprint
(Pdb) pprint.pprint(targets[name]["topology_outputs"])
{u'dummy_res': [{u'changed': False,
                 u'dummy_file': u'/tmp/dummy.hosts',
                 u'failed': False,
                 u'hosts': [u'web-fde56f-0.example.net']},
                {u'changed': False,
                 u'dummy_file': u'/tmp/dummy.hosts',
                 u'failed': False,
                 u'hosts': [u'web2-fde56f-0.example.net',
                            u'web2-fde56f-1.example.net']}],
u'aws_res': [{u'changed': true,
                    'xxx': xxx
                   }] # just for idea 

}
(Pdb) pprint.pprint(targets[name]["outputs"])
{u'resources': [{u'changed': False,
                 u'dummy_file': u'/tmp/dummy.hosts',
                 u'failed': False,
                 u'hosts': [u'web-fde56f-0.example.net']},
                {u'changed': False,
                 u'dummy_file': u'/tmp/dummy.hosts',
                 u'failed': False,
                 u'hosts': [u'web2-fde56f-0.example.net',
                            u'web2-fde56f-1.example.net']},
                {u'changed': False,
                 u'hosts': [u'ec2ip.x.x.x'] }                          # just for the idea
                  ]}
```
 However, topology_outputs consolidates everything into a single key resources regardless of their type. topology_outputs will be helpful for the future use cases when users would like to have only a specific cloud provider type of inventory. 
You can say on can identify the resource from "outputs" by writing up logic to identify schema of each output resource which seems a difficult tasks and hard to maintain since the output formats changes can break the code anytime. 
There might be a better way to do this . But adding a key felt like a easier way.

Reply:
@samvarankashyap Actually, I'm not saying that it isn't a separate item at all. I was just saying that since it's of 'output' type, it should be stored in the 'output' key. There can be *many* items in the output area, each of them can be different. I realize that resources and inventory (or topology outputs as you call them) are different. In fact, in another comment on this PR, I suggested a solution to separate out the 'topology' outputs from the 'resource' outputs. However, they are both outputs, just different kinds. Does this make better sense?